packages/davinci-client/src/lib/client.store.ts

```diff
@@ -190,11 +206,24 @@ export async function davinci<ActionType extends ActionTypes = ActionTypes>({
       }
 
       const { id } = collector;
-      const collectorToUpdate = nodeSlice.selectors.selectCollector(store.getState(), id);
+      const { error, state: collectorToUpdate } = nodeSlice.selectors.selectCollector(
+        store.getState(),
+        id,
+      );
+
+      if (error) {
+        log.error(error.message);
+        return function () {
+          return {
+            type: 'internal_error' as const,
+            error: { message: error.message, type: 'state_error' as const },
+          };
+        };
+      }
 
       if (!collectorToUpdate) {
         return function () {
-          console.error('Collector not found');
+          log.error('Collector not found');
```

Comment:
Also here we are logging inside of the returned function but above we are logging before the returned function.